orderapi/src/main/java/fi/hel/verkkokauppa/order/service/accounting/AccountingSlipService.java

```diff
@@ -0,0 +1,201 @@
+        return accountingSlips;
+    }
+
+    public Map<String, List<String>> groupAccountingsByDate(List<Order> ordersToAccount) {
+        Map<String, List<String>> map = new HashMap<>();
+
+        List<String> orderIds = ordersToAccount.stream()
+                .map(Order::getOrderId)
+                .collect(Collectors.toList());
+
+        List<OrderAccounting> orderAccountings = orderAccountingService.getOrderAccountings(orderIds);
+
+        for (OrderAccounting orderAccounting : orderAccountings) {
+            String createdAt = orderAccounting.getCreatedAt();
+
+            if (LocalDate.parse(createdAt).isBefore(LocalDate.now())) {
+                List<String> accountingsForDate = map.get(createdAt);
+
+                if (accountingsForDate == null) {
+                    ArrayList<String> accountings = new ArrayList<>();
+                    accountings.add(orderAccounting.getOrderId());
+                    map.put(createdAt, accountings);
+                } else {
+                    accountingsForDate.add(orderAccounting.getOrderId());
+                }
+            }
+        }
+
+        return map;
+    }
+
+    private AccountingSlipDto createAccountingSlipForDate(Map.Entry<String, List<String>> accountingsForDate) {
+        List<OrderItemAccountingDto> itemAccountings = getOrderItemAccountingsForDate(accountingsForDate);
+
+        // Summataan hinnat yksittäisen tiliöinnin tasolle
+        Collection<OrderItemAccountingDto> summedItemAccountings = itemAccountings.stream()
+                .collect(Collectors.toMap(OrderItemAccountingDto::createKey, Function.identity(), (OrderItemAccountingDto::merge)))
+                .values();
+
+        String postingDate = accountingsForDate.getKey();
+        String documentDate = DateTimeUtil.getDate();
+        String accountingSlipId = UUIDGenerator.generateType3UUIDString(postingDate, documentDate);
+
+        List<AccountingSlipRowDto> rows = new ArrayList<>();
+        int rowNumber = 1;
+
+        for (OrderItemAccountingDto summedItemAccounting : summedItemAccountings) {
+            String accountingSlipRowId = UUIDGenerator.generateType3UUIDString(accountingSlipId, Integer.toString(rowNumber));
+
+            AccountingSlipRowDto accountingSlipRowDto = new AccountingSlipRowDto(
+                    accountingSlipRowId,
+                    accountingSlipId,
+                    formatSum(summedItemAccounting.getPriceGrossAsDouble()),
+                    formatSum(summedItemAccounting.getPriceNetAsDouble()),
+                    summedItemAccounting.getMainLedgerAccount(),
+                    summedItemAccounting.getVatCode(),
+                    summedItemAccounting.getInternalOrder(),
+                    summedItemAccounting.getProfitCenter(),
+                    summedItemAccounting.getProject(),
+                    summedItemAccounting.getOperationArea()
+            );
+
+            rows.add(accountingSlipRowDto);
+            rowNumber++;
+        }
+
```

Comment:
what state is this in, a sane default would be either a valid value, a known empty value or bit more explanation of what the expected fill value is like